[dev-v2.12] longhorn 107.0.0+up1.8.2 add

mantissahz opened the pull request:
Release Longhorn v1.8.2 into Rancher 2.12

Longhorn https://github.com/longhorn/longhorn/issues/10912

Signed-off-by: James Lu [james.lu@suse.com](mailto:james.lu@suse.com)

cc @PhanLe1010 @innobead @derekbit @yangchiu

derekbit commented:
cc @PhanLe1010 for review

PhanLe1010 reviewed (commented):
I see that we are removing `107.0.0+up1.9.1` charts and replace it by the correct version `107.1.0+up1.9.1`. Can be problematic for user who already install `107.0.0+up1.9.1`, @nicholasSUSE ?

nicholasSUSE reviewed (changes requested):
You should never remove an already released version:  
https://github.com/rancher/charts/wiki/0%E2%80%90Chart-Owners#removing-charts

You can hide the version, but I strongly advice against doing that at the same time as releasing the new version. 

You should release the new version first and, 
 
Wait for your users to update and then hide the old version.

PhanLe1010 commented:
Thanks @nicholasSUSE 

HI @mantissahz Can we keep the `107.0.0+up1.9.1` version and release the new version `107.1.0+up1.9.1`?
We can hide the `107.0.0+up1.9.1` version later afer this PR. 

Thank you

mantissahz commented:
Hi @nicholasSUSE, @PhanLe1010,

I have updated the PR to keep the 107.0.0+up1.9.1 version and release the new version 107.1.0+up1.9.1.
Thanks for the advice.

PhanLe1010 reviewed (commented):
Do you know why the package file still mentions `version: 107.0.0`? @mantissahz 

https://github.com/rancher/charts/blob/9be66f024c4df65e19bb90717596ebad7473e57d/packages/longhorn/longhorn/longhorn-1.9/package.yaml

mantissahz commented:
>Do you know why the package file still mentions version: 107.0.0? @mantissahz
https://github.com/rancher/charts/blob/9be66f024c4df65e19bb90717596ebad7473e57d/packages/longhorn/longhorn/longhorn-1.9/package.yaml

I think the package version change was in the commit `Remove 107.0.0+up1.9.1`, and it was back to 107.0.0 when I removed this commit.

Corrected now.

PhanLe1010 reviewed (approved):
LGTM

roger-ryao commented:
Hi @mantissahz  @PhanLe1010 
Test Conditions:

Rancher prime: `v2.12.1` &  Rancher : `v2.12.1`

Verify is passed as follows step
- [x] Fresh installation
- [x] Uninstallation
